src/connPoolUDP.h

```diff
@@ -0,0 +1,249 @@
+    This class helps handling a UDP connection to another machine, in a way that different classes can use the same connection for a purpose they specify.
+    
+    Incoming data is read once, and any method that has subcribed beforehand to get the message, will receive it as a signal. 
+    If a method is not interested in the connection anymore, it can unsubscribe from the messages listened, or even prompt the pool class to close the connection. 
+    But a connection is only closed if the number of its subscribed "listener" methods is 0. 
+    
+    This class makes it very easy to manage when for example PupilEXT is running and listening to e.g. a UDP port, and gets a remote command
+    from that port to start streaming (on that same port, or any other port). Also it can be used to manage listening to the same port (pool instance)
+    from different classes, with no need to worry for failing to open an already opened port.
+
+    For sending data, or accessing socket directly, the socket instance pointer can be gotten from outside this class, via getInstance().
+    This should be used with caution, as the connection should not be closed directly, but only via closeConnection() method of the pool class.
+
+    IMPORTANT:
+        ! The PORTNUMBER field is used as a unique key for identifying a UDP port. (It could be the PORTNUMBER + IPADDRESS combo, but this is easier.)
+        When this class is told to open a new connection (and assign an instance) to a UDP port,
+            1, whose PORTNUMBER matches that of an already opened port /existing instance:
+                - if the IPADDRESS (filter) setting is not matching with the desired one, it will:
+                    - not open a new port and assign a pool UDP instance,
+                    - and neither return with the identifier (vector index) of the matching instance.
+                    => Connection will be seen as unsuccessful from the method that called setupAndOpenConnection().
+                - if the IPADDRESS (filter) setting is also the same, it will follow a conservative strategy:
+                    - not open a new port and assign a pool COM instance,
+                    - but return with the identifier (vector index) of the matching instance.
+                    => Connection will be seen as successful from the method that called setupAndOpenConnection().
+            2, whose PORTNUMBER does NOT match with the desired one, then it will:
+                    - open a new port and assign a pool UDP instance.
+
+        One opened UDP port /pool UDP instance can be used for multiple purposes, marked by purposeFlags.
+        Also - in this version - multiple ports can be used for the same purpose.
+        Classes that use these pools must internally keep track of their pool instance IDs assigned, returned by 
+        setupAndOpenConnection(), and reset it to e.g. -1 or other invalid value when the connection is closed.
+*/
+class ConnPoolUDP : public QObject {
```

Comment:
This class should track if its connections are open and perform a check and close in destructor to make sure everything is cleaned up.
Also, it shouldn't all be in the header file.